test/fbamodel_importer_test.py

```diff
@@ -0,0 +1,177 @@
+
+    def getWsName(self):
+        if hasattr(self.__class__, 'wsName'):
+            return self.__class__.wsName
+        suffix = int(time.time() * 1000)
+        wsName = "test_kb_uploadmethods_" + str(suffix)
+        ret = self.getWsClient().create_workspace({'workspace': wsName})  # noqa
+        self.__class__.wsName = wsName
+        return wsName
+
+    def getImpl(self):
+        return self.__class__.serviceImpl
+
+    def getContext(self):
+        return self.__class__.ctx
+
+    def mock_download_staging_file(params):
+        print 'Mocking DataFileUtilClient.download_staging_file'
+        print params
+
+        fq_filename = params.get('staging_file_subdir_path')
+        fq_path = os.path.join('/kb/module/work/tmp', fq_filename)
+        shutil.copy(os.path.join("data", fq_filename), fq_path)
+
+        return {'copy_file_path': fq_path}
+
+    def test_bad_as_media_from_staging(self):
+        invalid_params = {
+            'file_type': 'sbml',
+            'workspace_name': self.getWsName(),
+            'fba_model_name': 'MyModel'
+        }
+        with self.assertRaisesRegexp(
+                    ValueError, 'Required parameter "model_file" is missing'):
+            self.getImpl().import_file_as_fba_model_from_staging(
+                self.getContext(), invalid_params)
+
+        invalid_params = {
+            'model_file': 'test_model-reactions.tsv',
+            'file_type': 'tsv',
+            'workspace_name': self.getWsName(),
+            'fba_model_name': 'MyModel'
+        }
+        with self.assertRaisesRegexp(
+                ValueError, 'A compound file is required for tsv upload.'):
+            self.getImpl().import_file_as_fba_model_from_staging(
+                self.getContext(), invalid_params)
+
+        invalid_params = {
+            'model_file': 'test_model-reactions.tsv',
+            'file_type': 'csv',
+            'workspace_name': self.getWsName(),
+            'fba_model_name': 'MyModel'
+        }
+        with self.assertRaisesRegexp(
+                ValueError, '"csv" is not a valid import file_type'):
+            self.getImpl().import_file_as_fba_model_from_staging(
+                self.getContext(), invalid_params)
+
+    @patch.object(DataFileUtil, "download_staging_file", side_effect=mock_download_staging_file)
+    def test_import_as_media_from_staging(self, download_staging_file):
+
+        params = {
+          'model_file': 'test_model.sbml',
+          'file_type': 'sbml',
+          'workspace_name': self.getWsName(),
+          'fba_model_name': 'MyModel'
+        }
+
+        ref = self.getImpl().import_file_as_fba_model_from_staging(
+            self.getContext(), params)
+        self.assertTrue('obj_ref' in ref[0])
+        self.assertTrue('report_ref' in ref[0])
+        self.assertTrue('report_name' in ref[0])
+
+        params = {
+            'model_file': 'test_model.xlsx',
+            'file_type': 'excel',
+            'genome': '7601/4/1',
```

Comment:
do something like https://github.com/kbaseapps/kb_stringtie/blob/master/test/kb_stringtie_server_test.py#L82-L92

we should never use real object. Reviewers might not have permissions. And might test it on different environment.